### ISSUE TITLE
Pen Eraser Functionality

### ISSUE THREAD

**Issue by Squalmals** [zsviczian/obsidian-excalidraw-plugin#783]
I am using an HP laptop with a touchscreen and pen. The pen has two buttons which are able to be rebound, but by default they are set to the "eraser" option, which works in apps like paint, OneNote, Word, Krita, and many others by holding the button and allowing the pen to function as an eraser while the button is held.

It would be very nice if in Excalidraw this functionality would be preserved, as right now if I want to erase something I have to switch to the eraser tool, erase something, and then switch back which is quite annoying and disrupts the workflow, especially with the location of the eraser button. I can rebind the button on the pen to "E" which would allow for switching to the eraser easier, but then that doesn't solve the issue of needing to still go all the way back to the toolbar to change back to the pen tool.

Overall it's not that big of an issue, but it is an annoyance that would substantially improve the flow of the application for pen users if it were fixed.

The pen I am using is the HP MPP 2.0 tilt pen that was included with my laptop. I will include a screenshot of the pen software as well.

Overall I am very much enjoying using excalidraw in obsidian but I feel the experience would be drastically improved if this issue were resolved. I am also open to workarounds if anyone knows one.

![image](https://user-images.githubusercontent.com/81203469/187988925-71a0d8d7-5001-4065-b8eb-997e04159006.png)

**Comment by zsviczian** [comment]
> I can rebind the button on the pen to "E" which would allow for switching to the eraser easier, but then that doesn't solve the issue of needing to still go all the way back to the toolbar to change back to the pen tool.

When you press "e" the second time Excalidraw switches back to the last tool used. I just tested it and it seems to work for me. Is this not your experience?

**Comment by Squalmals** [comment]
> > I can rebind the button on the pen to "E" which would allow for switching to the eraser easier, but then that doesn't solve the issue of needing to still go all the way back to the toolbar to change back to the pen tool.
> 
> When you press "e" the second time Excalidraw switches back to the last tool used. I just tested it and it seems to work for me. Is this not your experience?

This is true, I must've missed this, however apparently the pen software I use that came with my laptop apparently does not actually allow for rebinding to custom key presses, only to a very limited list of functions built into the software. Because of this, my issue still stands, unfortunately.

**Comment by rorar** [comment]
@zsviczian here's a potential fix:
https://github.com/excalidraw/excalidraw/issues/5281
(scroll down to the latest comment)

**Comment by RelativeProgramming** [comment]
Same problem here. I'm using Obsidian on an Android tablet (Samsung Galaxy Tab S7) and would like to use the pen button for the eraser functionality, like it is done in all the other note taking apps.

**Comment by zsviczian** [comment]
@rorar, thanks! I will try to play with this... frankly I did not notice your response until @RelativeProgramming just commented on this issue.

This looks like a reasonable solution. I will try to play with this.... however I also think that you could test out the recommended solution as well. You can use Tempalter (or the Developer Console in Obsidian) to inject the recommended script and see if it solves the issue. 

@RelativeProgramming - whether this will work for the sPen I can't say. The problem is that both the Apple Pencil and the sPen require dedicated SDKs to be compiled with the native mobile app. Obsidian is not a native mobile app but runs in a Web View component (a web browser wrapped into an app), and as such, it is not possible to use these SDKs. That said, if I find time I will experiment a little with my sPen to see if I can find a solution for Samsung devices.

**Comment by georgeslanglois** [comment]
> [@RelativeProgramming](https://github.com/RelativeProgramming) - whether this will work for the sPen I can't say. The problem is that both the Apple Pencil and the sPen require dedicated SDKs to be compiled with the native mobile app. Obsidian is not a native mobile app but runs in a Web View component (a web browser wrapped into an app), and as such, it is not possible to use these SDKs. That said, if I find time I will experiment a little with my sPen to see if I can find a solution for Samsung devices.

Hello my man, 
I was just wondering if some solution popped out 3 years later this topic 😅

**Comment by caiopb** [comment]
> > [@RelativeProgramming](https://github.com/RelativeProgramming) - whether this will work for the sPen I can't say. The problem is that both the Apple Pencil and the sPen require dedicated SDKs to be compiled with the native mobile app. Obsidian is not a native mobile app but runs in a Web View component (a web browser wrapped into an app), and as such, it is not possible to use these SDKs. That said, if I find time I will experiment a little with my sPen to see if I can find a solution for Samsung devices.
> 
> Hello my man, I was just wondering if some solution popped out 3 years later this topic 😅

No man I just landed here with my tab S10 FE seeking for this request. The s-pen still doesnt have the lateral button assigned to the eraser, which draws me back, since I got used to it already.

**Comment by zsviczian** [comment]
Consolidating pen / stylus / handwriting / drawing tablet support discussion into #2773.

Closing this as a duplicate so related requests stay centralized. As noted in #2773, hardware-specific pen features and behavior are now tracked there.
